tests/unit/lms/services/canvas_api/public_test.py

```diff
@@ -0,0 +1,238 @@
+"""
+Tests for the Canvas public API client.
+
+These tests take an integration style approach, rather than the mocked approach
+for the sub-components. The only thing mocked here is the requests session, the
+rest is the real deal.
+
+This is to prevent these tests from just becoming an MD5 of the code, and to
+actually show what calls we are supposed to be making in the end, rather than
+just what we feed into the system.
+"""
+from unittest.mock import sentinel
+
+import pytest
+from h_matchers import Any
+
+from lms.services import CanvasAPIAccessTokenError, CanvasAPIError, CanvasAPIServerError
+from lms.services.canvas_api import CanvasAPIClient
+
+
+class TestCanvasAPIClientGetToken:
+    # This is the only test where we fake out the underlying class, because
+    # this _one_ call is just a pass through.
```

Comment:
Although hilariously the very first test, actually _does_ need to mock out the `AuthenticatedClient`...